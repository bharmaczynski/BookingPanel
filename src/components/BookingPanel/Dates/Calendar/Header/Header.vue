<template>
  <div class="header">
    <div
        class="header__arrow header__arrow--previous"
        @click="$emit('previousArrowClicked')"
    >
      <ArrowLeft />
    </div>
    <span class="header__label">{{ headerLabel }}</span>
    <div
        class="header__arrow header__arrow--next"
        @click="$emit('nextArrowClicked')"
    >
      <ArrowRight />
    </div>
  </div>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import ArrowLeft from '@/assets/icons/left.svg';
import ArrowRight from '@/assets/icons/right.svg';

@Component({
  components: {
    ArrowLeft,
    ArrowRight,
  }
})
export default class Header extends Vue {
  @Prop({ required: true})
  headerLabel!: string;
}
</script>
<style lang="scss" scoped>
.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 10px;
  height: 60px;
  background: $downy;
  font-size: 18px;

  &__label {
    color: #fff;
  }

  &__arrow {
    width: 26px;
    height: 26px;
    padding: 5px;
    cursor: pointer;

    svg {
      width: auto;
      height: 100%;

      path {
        transition: fill 0.2s ease;
      }
    }

    &:hover {
      svg {
        path {
          fill: #fff;
        }
      }
    }
  }
}

</style>
