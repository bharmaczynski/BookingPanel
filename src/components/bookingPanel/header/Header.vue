<template>
    <div class="header">
        <span class="header__price text--bold font-size-xxl"> {{ price }} z≈Ç </span>
        <Rating :numberOfRates="numberOfRates" :rating="rating" />
    </div>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import Rating from '@/components/bookingPanel/rating/Rating.vue';

@Component({
    components: {
        Rating,
    },
})
export default class Header extends Vue {
    @Prop({ required: true })
    price!: number;

    @Prop({ default: 0 })
    numberOfRates!: number;

    @Prop({ default: 0 })
    rating!: number;
}
</script>
