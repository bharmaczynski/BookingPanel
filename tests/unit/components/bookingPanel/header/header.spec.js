import { shallowMount } from '@vue/test-utils';
import Header from '@/components/bookingPanel/header/Header.vue';

describe('Header', () => {
    it('should match initial snapshot', () => {
        const wrapper = shallowMount(Header);

        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it('should match snapshot with props', () => {
        const wrapper = shallowMount(Header, {
            propsData: {
                price: 100,
                numberOfRates: 43,
                rating: 4.3,
            },
        });

        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});
