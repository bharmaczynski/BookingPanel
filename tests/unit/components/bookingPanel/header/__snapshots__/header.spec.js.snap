// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match initial snapshot 1`] = `
<div
  class="header"
>
  <span
    class="header__price text--bold font-size-xxl"
  >
     0 zł 
  </span>
   
  <rating-stub
    numberofrates="0"
    rating="0"
  />
</div>
`;

exports[`Header should match snapshot with props 1`] = `
<div
  class="header"
>
  <span
    class="header__price text--bold font-size-xxl"
  >
     100 zł 
  </span>
   
  <rating-stub
    numberofrates="43"
    rating="4.3"
  />
</div>
`;
